hyperspy/_signals/sed.py

```diff
@@ -0,0 +1,465 @@
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import division
+
+import traits.api as t
+import math
+import numpy as np
+import scipy.ndimage as ndi
+
+from hyperspy._signals.signal2d import Signal2D
+from hyperspy.decorators import only_interactive
+from hyperspy.gui.sed import SEDParametersUI
+from hyperspy.defaults_parser import preferences
+import hyperspy.gui.messages as messagesui
+
+
+class SEDPattern(Signal2D):
+    _signal_type = "SED_Pattern"
+
+    def __init__(self, *args, **kwargs):
+        Signal2D.__init__(self, *args, **kwargs)
+        # Attributes defaults
+        if 'Acquisition_instrument.TEM' not in self.metadata:
+            if 'Acquisition_instrument.SEM' in self.metadata:
+                self.metadata.set_item(
+                    "Acquisition_instrument.TEM",
+                    self.metadata.Acquisition_instrument.SEM)
+                del self.metadata.Acquisition_instrument.SEM
+        self._set_default_param()
+
+    def _set_default_param(self):
+        """Set microscope parameters to default values (defined in preferences)
+        """
+
+        md = self.metadata
+        md.Signal.signal_type = 'SED_Pattern'
+
+        if "Acquisition_instrument.TEM.beam_energy" not in md:
+            md.set_item("Acquisition_instrument.TEM.beam_energy",
+                        preferences.SED.sed_beam_energy)
+        if "Acquisition_instrument.TEM.camera_length" not in md:
+            md.set_item("Acquisition_instrument.TEM.camera_length",
+                        preferences.SED.sed_camera_length)
+        if "Acquisition_instrument.TEM.scan_rotation" not in md:
+            md.set_item("Acquisition_instrument.TEM.scan_rotation",
+                        preferences.SED.sed_scan_rotation)
+        if "Acquisition_instrument.TEM.convergence_angle" not in md:
+            md.set_item("Acquisition_instrument.TEM.convergence_angle",
+                        preferences.SED.sed_convergence_angle)
+        if "Acquisition_instrument.TEM.precession_angle" not in md:
+            md.set_item("Acquisition_instrument.TEM.precession_angle",
+                        preferences.SED.sed_precession_angle)
+        if "Acquisition_instrument.TEM.precession_frequency" not in md:
+            md.set_item("Acquisition_instrument.TEM.precession_frequency",
+                        preferences.SED.sed_precession_frequency)
+        if "Acquisition_instrument.TEM.Detector.SED.exposure_time" not in md:
+            md.set_item("Acquisition_instrument.TEM.Detector.SED.exposure_time",
+                        preferences.SED.sed_exposure_time)
```

Comment:
Most of these entries are not in the [metadata specification](http://hyperspy.org/hyperspy-doc/current/user_guide/metadata_structure.html). The specifications should be expanded to include the ones strictly needed. Why is there a `scan_rotation` and `sed_scan_rotation`?

Wouldn't `Acquisition_instrument.TEM.Detector.diffraction.exposure` be better than `Acquisition_instrument.TEM.Detector.SED.exposure_time`? All quantities should include units in brackets e.g. "exposure (s)".